craft_parts/plugins/base.py

```diff
@@ -242,11 +242,16 @@ def _get_rewrite_shebangs_commands(self) -> list[str]:
         shebangs in the final environment should be handled.
         """
         script_interpreter = self._get_script_interpreter()
+        find_cmd = (
+            f'find "{self._part_info.part_install_dir}" -type f -executable -print0'
+        )
+        xargs_cmd = "xargs --no-run-if-empty -0"
```

Comment:
did a little refactoring here because it was getting long and hard to read; the `--no-run-if-empty` bit is the new one

Reply:
Thanks both for doing that (the new code is way easier to read) and for the comment noting the change!